dumb-init.c

```diff
@@ -23,21 +24,46 @@
     } \
 } while (0)
 
-pid_t child = -1;
+pid_t child_pid = -1;
 char debug = 0;
 char use_process_group = 1;
 
 void signal_handler(int signum) {
     DEBUG("Received signal %d.\n", signum);
 
-    if (child > 0) {
-        kill(use_process_group ? -child : child, signum);
+    if (child_pid > 0) {
+        kill(use_process_group ? -child_pid : child_pid, signum);
         DEBUG("Forwarded signal to child.\n");
     } else {
         DEBUG("Didn't forward signal, no child exists yet.");
     }
 }
 
+void reap_zombies(int signum) {
+    /*
+     * As PID 1, dumb-init is expected to handle reaping of zombie processes.
+     *
+     * If a process's parent exits, the child is orphaned and its new parent is
+     * PID 1. If that child later exits, it becomes a zombie process until its
+     * parent (now dumb-init) calls wait() on it.
```

Comment:
note: we don't simply use SIGDFL because we want the exit code of our direct subprocess.